File: windows/particles/OpenBurnBar.Particles/Substrates/Families/FlowFamily.cs
using OpenBurnBar.Particles.Model;
using OpenBurnBar.Particles.Substrates.Flow;

namespace OpenBurnBar.Particles.Substrates.Families;

/// <summary>
/// Flow Field family registry — C# port of Swift <c>Families/FlowFamily.swift</c>.
/// Directional / streaming motion fields: bioluminescent wake, liquid-glass ribbon,
/// calligraphic streamlines, drifting petals. Ids / labels / accents mirror the Swift
/// descriptors byte-for-byte.
/// </summary>
public static class FlowFamily
{
    public static readonly SubstrateDescriptor[] Styles =
    {
        new SubstrateDescriptor(
            id: "flow.plankton-wake", family: SubstrateFamily.Flow,
            label: "Plankton Wake", hint: "bioluminescence",
            accent: new Rgba(86 / 255.0, 198 / 255.0, 224 / 255.0),
            accent2: new Rgba(120 / 255.0, 255 / 255.0, 210 / 255.0),
            make: () => new PlanktonWakeSubstrate()),
        new SubstrateDescriptor(
            id: "flow.glass-ribbon", family: SubstrateFamily.Flow,
            label: "Glass Ribbon", hint: "ribbon",
            accent: new Rgba(120 / 255.0, 210 / 255.0, 235 / 255.0),
            accent2: new Rgba(180 / 255.0, 245 / 255.0, 245 / 255.0),
            make: () => new GlassRibbonSubstrate()),
        new SubstrateDescriptor(
            id: "flow.silk-streamline", family: SubstrateFamily.Flow,
            label: "Silk Streamline", hint: "streamline",
            accent: new Rgba(100 / 255.0, 200 / 255.0, 224 / 255.0),
            accent2: new Rgba(150 / 255.0, 236 / 255.0, 232 / 255.0),
            make: () => new SilkStreamlineSubstrate()),
        new SubstrateDescriptor(
            id: "flow.petal-drift", family: SubstrateFamily.Flow,
            label: "Petal Drift", hint: "petals",
            accent: new Rgba(1.0, 150 / 255.0, 190 / 255.0),
            accent2: new Rgba(1.0, 200 / 255.0, 170 / 255.0),
            make: () => new PetalDriftSubstrate()),
    };
}

File: OpenBurnBarCore/Sources/OpenBurnBarUI/Views/Substrate/Families/FlowFamily.swift
import Foundation
import OpenBurnBarKernel

/// Flow Field family substrate registry. Ported from imaginethat
/// `glyph/stage/styles/flow/`.
enum FlowFamily {
    static let styles: [SubstrateDescriptor] = [
        SubstrateDescriptor(
            id: "flow.plankton-wake", family: .flow,
            label: "Plankton Wake", hint: "bioluminescence",
            accent: RGBA(r: 86/255, g: 198/255, b: 224/255), accent2: RGBA(r: 120/255, g: 255/255, b: 210/255),
            make: { PlanktonWakeSubstrate() }),
        SubstrateDescriptor(
            id: "flow.glass-ribbon", family: .flow,
            label: "Glass Ribbon", hint: "ribbon",
            accent: RGBA(r: 120/255, g: 210/255, b: 235/255), accent2: RGBA(r: 180/255, g: 245/255, b: 245/255),
            make: { GlassRibbonSubstrate() }),
        SubstrateDescriptor(
            id: "flow.silk-streamline", family: .flow,
            label: "Silk Streamline", hint: "streamline",
            accent: RGBA(r: 100/255, g: 200/255, b: 224/255), accent2: RGBA(r: 150/255, g: 236/255, b: 232/255),
            make: { SilkStreamlineSubstrate() }),
        SubstrateDescriptor(
            id: "flow.petal-drift", family: .flow,
            label: "Petal Drift", hint: "petals",
            accent: RGBA(r: 1.0, g: 150/255, b: 190/255), accent2: RGBA(r: 1.0, g: 200/255, b: 170/255),
            make: { PetalDriftSubstrate() })
    ]
}
